Add reduced representation of genes

cmdcolin opened the pull request:
Adds a couple features for reduced representation of genes via "displayMode" on PileupRenderer and SvgFeatureRenderer

Also renames alignmentHeight to just height

The displayMode code here has to do some work in various parts of the code here, it's be nice if it was a little more hands-off, but it does accomplish jbrowse 1 compatibility

rbuels commented on packages/core/util/layouts/GranularRectLayout.ts:
```diff
@@ -306,12 +306,15 @@ export default class GranularRectLayout<T> implements BaseLayout<T> {
     pitchX: number | undefined
     pitchY: number | undefined
     maxHeight: number | undefined
+    displayMode: string | undefined
```
is there some way to add a more restrictive type here that restricts the displayMode to undefined, 'normal', or 'collapsed'?

cmdcolin replied:
typescript can handle this type of thing, ya. I guess it just depends on, for some types of things like this, whether you get benefit from the type system by being that granular since of course it's not really a run time thing

rbuels commented:
this looks fine in principal, any reason not to merge once build is passing?

cmdcolin commented:
Probably good to merge after gettin it buildin

cmdcolin commented:
Well,, I'll actually do a once over...I feel like this PR was a little intrusive on the process

cmdcolin commented:
I added some tests for both compact and reducedRepresentation and renamed squish to reducedRepresentation (squish while fun seemed vague?)

My concern about maybe making this code "less obtrusive" may probably require some more general refactoring of svg feature. I had a sort of misconception about the height but there is a nice getFeatureHeight in JB1 that applies the height changes, not really a equivalent here

rbuels commented on packages/alignments/src/PileupRenderer/configSchema.js:
```diff
@@ -12,6 +13,16 @@ export default ConfigurationSchema(
+        'compact',
+        'collapse',
+      ]),
+      description: 'Alternative display modes',
```
can we have a more descriptive description
